src/rust/cryptography-x509/src/common.rs

```diff
@@ -112,6 +120,11 @@ pub enum AlgorithmParameters<'a> {
     #[defined_by(oid::DSA_WITH_SHA512_OID)]
     DsaWithSha512(Option<asn1::Null>),
 
+    #[defined_by(oid::DH_OID)]
+    Dh(DHParams<'a>),
+    #[defined_by(oid::DH_KEY_AGREEMENT_OID)]
+    DhKeyAgreement(DHParams<'a>),
```

Comment:
I am not confident either of these are right, I'm not sure how to verify further.